lib/ABI/RegisterStateDeductions.cpp

```diff
@@ -457,7 +457,7 @@ struct DeductionImpl {
         return abi::RegisterState::No;
       } else {
         using namespace abi::RegisterState;
-        if (Result != Maybe && Result != No) {
+        if (Result != Maybe && Result != No && Result != NoOrDead) {
```

Comment:
I small improvement on how we should handle `NoOrDead` registers.

https://github.com/revng/revng/pull/232/commits/32f695af7208d07f671a76bf145dc22dc82ca44d

@aleclearmind, feel free to close if the commit looks good.

Reply:
`ABI: let NoOrDead painlessly decay to No`